pkg/integrations/configrequest/protocol/v1.go

```diff
@@ -0,0 +1,47 @@
+package protocol
+
+import (
+	"fmt"
+
+	"github.com/newrelic/infrastructure-agent/pkg/databind/pkg/databind"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/track/ctx"
+	"github.com/newrelic/infrastructure-agent/pkg/integrations/v4/config"
+)
+
+type v1 struct {
+	Action     string `json:"action"`
+	ConfigName string `json:"config_name"`
+	Config     struct {
+		Databind     databind.YAMLAgentConfig `json:",inline"`
```

Comment:
This feels a bit wrong :) Would it make sense to decouple the serialization format from the config object? ie. having one AgentConfig object and then one YAML and one JSON that derive from it.

Reply:
Yes the naming is quite confusing here, but i would remove the `YAML` prefix of the package since serialization format is described in the object annotations and both can live together. @cristianciutea wdyt? if ok i could do it in a separate PR.

Reply:
that sounds good to me